DSC/dsc.py

```diff
@@ -267,29 +288,28 @@ def install_dsc_packages():
     dsc_package_path = dsc_package_prefix + openssl_version
     waagent.AddExtensionEvent(name=ExtensionShortName, op='InstallInProgress', isSuccess=True, message="Installing omipackage version: " + omi_package_path + "; dsc package version: " +  dsc_package_path)
     if distro_category == DistroCategory.debian:
-        deb_install_pkg(omi_package_path + '.x64.deb', 'omi', omi_version_deb)
-        deb_install_pkg(dsc_package_path + '.x64.deb', 'dsc', dsc_version_deb )
+        deb_install_pkg(omi_package_path + '.x64.deb', 'omi', omi_major_version, omi_minor_version, omi_build, omi_release)
+        deb_install_pkg(dsc_package_path + '.x64.deb', 'dsc', dsc_major_version, dsc_minor_version, dsc_build, dsc_release)
     elif distro_category == DistroCategory.redhat or distro_category == DistroCategory.suse:
-        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi', omi_major_version, omi_minor_version, omi_build)
-        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc', dsc_major_version, dsc_minor_version, dsc_build)
+        rpm_install_pkg(omi_package_path + '.x64.rpm', 'omi', omi_major_version, omi_minor_version, omi_build, omi_release)
+        rpm_install_pkg(dsc_package_path + '.x64.rpm', 'dsc', dsc_major_version, dsc_minor_version, dsc_build, dsc_release)
 
-def compare_pkg_version(system_package_version, major_version, minor_version, build):
-    version = re.match('(\d+).(\d+).(\d+)', system_package_version)
-    if version is not None and ((int(version.group(1)) > major_version) or (int(version.group(1)) == major_version and int(version.group(2)) > minor_version) or (int(version.group(1)) == major_version and int(version.group(2)) == minor_version and int(version.group(3)) >= build)):
+def compare_pkg_version(system_package_version, major_version, minor_version, build, release):
+    version = re.match('(\d+).(\d+).(\d+).(\d+)', system_package_version)
```

Comment:
regex pattern is same as on line 247. Should prolly be a class/script level string